.github/workflows/publish.yml

```diff
@@ -0,0 +1,108 @@
+        uses: lewagon/wait-on-check-action@ccfb013c15c8afb7bf2b7c028fb74dc5a068cccc # v1.3.4
+        with:
+          ref: ${{ github.ref }}
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          wait-interval: 15
+          running-workflow-name: Wait for CI
+          verbose: true
+          # Don't specify check-name or check-regexp to wait for ALL checks
+          # except the current workflow (specified by running-workflow-name)
+
+  publish:
+    name: Publish to Gradle Plugin Portal
+    runs-on: ubuntu-latest
+    needs: wait-for-ci
+    timeout-minutes: 30
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v4
+        with:
+          java-version: 11
+          distribution: temurin
+
+      - name: Cache Gradle
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: gradle-publish-${{ runner.os }}-${{ hashFiles('**/*.gradle*','**/gradle-wrapper.properties') }}
+          restore-keys: |
+            gradle-publish-${{ runner.os }}-
+            gradle-${{ runner.os }}-
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@06832c7b30a0129d7fb559bcc6e43d26f6374244 # v4.3.1
+        with:
+          gradle-version: wrapper
+
+      - name: Extract version from tag or input
+        id: get_version
+        run: |
+          set -e
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+            if [[ -n "${{ github.event.inputs.tag }}" ]]; then
+              v="${{ github.event.inputs.tag }}"
+            else
+              git fetch --tags
+              v="$(git describe --tags --abbrev=0)"
+              v="${v#v}"
+            fi
+          else
+            v="${GITHUB_REF#refs/tags/}"
+            v="${v#v}"
+          fi
+          echo "VERSION=$v" >> $GITHUB_OUTPUT
+          echo "Extracted version: $v"
+
+      - name: Publish to Plugin Portal
+        env:
+          GRADLE_PUBLISH_KEY: ${{ secrets.GRADLE_PUBLISH_KEY }}
+          GRADLE_PUBLISH_SECRET: ${{ secrets.GRADLE_PUBLISH_SECRET }}
+        run: |
+          echo "Publishing version ${{ steps.get_version.outputs.VERSION }}"
+          # Original CircleCI command: final -P "release.version"="${CIRCLE_TAG}" --exclude-task test -P "gradle.publish.key"="${GRADLE_PUBLISH_KEY}" -P "gradle.publish.secret"="${GRADLE_PUBLISH_SECRET}"
+          # The -P flags for key/secret might not be strictly needed if your build.gradle(.kts)
+          # is set up to read them from environment variables (which ORG_GRADLE_PROJECT_ gradlePublishKey would become).
+          # Using -Prelease.version directly is fine.
+          ./gradlew publishPlugins \
+            -Prelease.version="${{ steps.get_version.outputs.VERSION }}" \
+            --no-daemon \
+            --exclude-task test
```

Comment:
note: im not for or against this versioning+publishing approach right now. i imagine we'll land on whatever is a modern versioning plugin + publishing as i havent worked closely in gradle in a few years and the best practice plugins have most likely changed.